src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,69 @@
+package duke.task;
+
+import duke.exception.TaskListOutOfBoundsException;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class TaskList {
+  private List<Task> tasks;
+
+  public TaskList() {
+    tasks = new ArrayList<>();
+  }
+
+  public Task getTask(int i) { // 1-indexed
+    try {
+      return tasks.get(i - 1);
+    } catch (IndexOutOfBoundsException e) {
+      throw new TaskListOutOfBoundsException(i);
+    }
+  }
+
+  public void addTask(Task t) {
+    tasks.add(t);
+  }
+
+  public String getNumberOfTasks() {
+    return String.format(
+        "Now you have %d task%s in the list.", tasks.size(), tasks.size() == 1 ? "" : "s");
```

Comment:
For clarity and consistency, consider renaming the method getNumberOfTasks() to something like getTaskCountMessage().